tools/mirror/src/chain_tracker.rs

```diff
@@ -121,7 +121,53 @@ struct NonceInfo {
 
 pub(crate) enum SentBatch {
     MappedBlock(TxBatch),
-    ExtraTxs(Vec<TargetChainTx>),
+    ExtraTxs(VecDeque<TargetChainTx>),
+}
+
+// We don't want to divide up a second into intervals that add up to us sending `tps` per second,
+// because the intervals will get too small. So sleep for at least 10 millis, and send possibly many per batch
+const MAX_BATCHES_PER_SEC: u32 = 100;
+
+struct TxBatchConfig {
+    batch_count: u32,
+    delay: Duration,
+    current_send_index: u32,
+    size_increase_index: u32,
+}
+
+// corresponds to the cadence set in the config, and dictates how we send transactions and how long we sleep for
+enum TrafficCadence {
+    MatchBlocks,
+    BlockInterval(Duration),
+    TPS(TxBatchConfig),
+}
+
+impl TrafficCadence {
+    fn new(tx_batch_interval: Option<Duration>, tps: Option<u32>) -> Self {
+        assert!(tx_batch_interval.is_none() || tps.is_none());
+        if let Some(tps) = tps {
+            let c = if tps <= MAX_BATCHES_PER_SEC {
+                TxBatchConfig {
+                    batch_count: 1,
```

Comment:
I think comment for batch_count would help, I do not understand why the batch is 1 here, do you mean everything within a second will be sent together or something else?